cdc/owner/feed_state_manager.go

```diff
@@ -562,6 +572,28 @@ func (m *feedStateManager) handleWarning(errs ...*model.RunningError) {
 		return
 	}
 	lastError := errs[len(errs)-1]
+
+	if m.state != nil && m.state.Status != nil {
+		currTime := m.upstream.PDClock.CurrentTime()
+		ckptTime := oracle.GetTimeFromTS(m.state.Status.CheckpointTs)
+		// Conditions:
+		// 1. checkpoint lag is large enough;
+		// 2. checkpoint hasn't been advanced for a long while;
+		// 3. the changefeed has been initialized.
+		if currTime.Sub(ckptTime) > defaultBackoffMaxElapsedTime &&
+			time.Since(m.checkpointTsAdvanced) > defaultBackoffMaxElapsedTime &&
+			m.state.Status.MinTableBarrierTs > m.state.Info.StartTs {
```

Comment:
`MinTableBarrierTs` cannot be used to indicate if the changefeed has finished incremental scanning, since it is equal to `ddlPuller` resolvedTs if there are no DDL events.